File: src/main/java/bad/robot/radiate/teamcity/HttpClientFactory.java
package bad.robot.radiate.teamcity;

import bad.robot.http.CommonHttpClient;
import bad.robot.http.HttpClient;

import static bad.robot.http.HttpClients.anApacheClient;
import static bad.robot.http.configuration.HttpTimeout.httpTimeout;
import static bad.robot.radiate.teamcity.BasicAuthConfiguration.basicAuthConfiguration;
import static com.google.code.tempusfugit.temporal.Duration.minutes;

public class HttpClientFactory {

    public HttpClient create(TeamCityConfiguration configuration) {
        CommonHttpClient client = anApacheClient();
        basicAuthConfiguration(configuration).applyTo(client);
        return client.with(httpTimeout(minutes(10)));
    }

}

File: src/main/scala/bad/robot/radiate/teamcity/HttpClientFactory.scala
package bad.robot.radiate.teamcity

import simplehttp.HttpClient
import simplehttp.HttpClients.anApacheClient
import simplehttp.configuration.HttpTimeout.httpTimeout
import bad.robot.radiate.config.Config
import com.google.code.tempusfugit.temporal.Duration.minutes

object HttpClientFactory {
  def apply() = new HttpClientFactory()
}

class HttpClientFactory private {
  def create(config: Config): HttpClient = {
    val client = anApacheClient
    Authentication(config).applyTo(client)
    client.`with`(httpTimeout(minutes(10)))
  }
}